### ISSUE TITLE
[Bug]: Unique tabs feature results in hidden tabs, confusing UX

### ISSUE THREAD

**Issue by stelcodes** [akinsho/bufferline.nvim#500]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

The recent change to enable unique tab names (PR #485) also hides tabs that are editing the same file. This is a significant, undocumented breaking change. It seems like this behavior was intended. When you have multiple tabs open that are editing the same file, only one is visible on the bufferline/tabline. Reproduction is simple, just set `mode = 'tabs'` and run `:tabnew somefile.txt` more than once.

I think it should be configurable and not the default behavior. I use tabs for the same file everyday and I'm sure I'm not the only one. This plugin loses most of its value for me if I can't see all the tabs that are open. I really don't see the value in hiding tabs because it ruins the visual aspect of `:tabnext` and `:tabprevious` which is the most common way of navigating through tabs. It becomes impossible to visualize your current tab position. It's easy to get in a state where there is no highlighted entry in the bufferline/tabline, and even if you `:tabnext` 10 times, there still may be no highlighted entry. To me that is very confusing behavior for a tabline.
 
![2022-08-11-15:27:30-screenshot](https://user-images.githubusercontent.com/22163194/184253157-c8909ed7-6ac2-46dd-9447-35a253973772.png)

### What did you expect to happen?

Before PR #485 was merged, you could open the same file for editing in multiple tabs and have all of them show up in the buffline/tabline
![2022-08-11-15:35:11-screenshot](https://user-images.githubusercontent.com/22163194/184253905-9e457ef0-c574-4ed3-8331-7d3ae75bc354.png)
.

### Config

```lua
  use {
    'akinsho/bufferline.nvim',
    -- commit = 'e164face76ddb589ba3493092c3ad7cf3afd98f0', -- right before unique tab pages PR (ok)
    -- commit = 'c4dd9b4de03b891f648b098c25e4dc1bc48a13e5', -- unique tab pages PR (not ok)
    config = function()
      local buff = require('bufferline')
      buff.setup {
        options = {
          mode = 'tabs',
        }
      }
    end
  }
```

### Additional Information

IMO buffer deduplication makes sense as a default but tab deduplication does not.

### commit

c4dd9b4de03b891f648b098c25e4dc1bc48a13e5

**Comment by akinsho** [comment]
@stelcodes the issue you describe is not by design, I believe the user who contributed the change also did not intend this. I'm not entirely clear why this is happening but if you look through the code that was contributed there is no specific intention to add this, so in all likelihood this is caused by some unforeseen interaction.

I haven't much time and a lot of issue across too many plugins so if anyone would like to contribute a fix that would be much appreciated, otherwise this will have to wait till some point in the future when I have time/motivation

**Comment by akinsho** [comment]
This should now be fixed by #501

**Comment by stelcodes** [comment]
@akinsho You rock!
